core/src/main/java/org/javarosa/core/io/StreamsUtil.java

```diff
@@ -7,10 +7,18 @@
 
 public class StreamsUtil {
 
-    private StreamsUtil() {
-        // private constructor
+    public static byte[] getStreamAsBytes(InputStream is) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        try {
+            StreamsUtil.writeFromInputToOutput(is, baos);
+            return baos.toByteArray();
+        } finally {
+            baos.close();
```

Comment:
That's fair, especially if we're buffering there is flushing behavior that can become relevant on close. We should probably keep an eye out for that kind of thing around the codebase in general.